packages/foundry/contracts/hooks/CommitMinerHook.sol

```diff
@@ -0,0 +1,119 @@
+        address pool,
+        uint256 amountIn,
+        uint256 amountOut,
+        address tokenIn,
+        address tokenOut,
+        address user
+    ) external override onlyVault {
+        bytes32 commitHash = keccak256(abi.encodePacked(block.timestamp, user, amountIn, amountOut, pool, gasleft()));
+        commitBacklog.push(Commit(commitHash, block.number, pool, user));
+
+        emit CommitGenerated(pool, user, commitHash, block.number);
+    }
+
+    // Randomness requester pays for a batch of commits using the specified paymentToken
+    function requestRandomness(uint256 commitCount, uint256 amountPaid) external nonReentrant {
+        require(commitCount <= commitBacklog.length, "Not enough commits available");
+
+        // Transfer payment from requester
+        require(paymentToken.transferFrom(msg.sender, address(this), amountPaid), "Payment transfer failed");
+
+        // Generate randomness based on the selected number of commits
+        bytes32 finalRandomness = aggregateCommits(commitCount);
+
+        // Emit event for randomness request
+        emit RandomnessRequested(msg.sender, amountPaid, commitCount);
+
+        // Distribute fees
+        distributeFees(amountPaid, commitCount);
+    }
+
+    // Function to aggregate commits into randomness
+    function aggregateCommits(uint256 commitCount) internal view returns (bytes32) {
+        bytes32 randomness;
+        for (uint256 i = commitBacklog.length - commitCount; i < commitBacklog.length; i++) {
+            randomness = keccak256(abi.encodePacked(randomness, commitBacklog[i].hash, block.number, gasleft()));
+        }
+        return randomness;
+    }
+
+    // Function to distribute fees among swappers, LPs, and the hook deployer
+    function distributeFees(uint256 amountPaid, uint256 commitCount) internal nonReentrant {
+        uint256 hookFee = (amountPaid * 10) / 100;  // 10% to hook deployer
+        uint256 swapperFee = (amountPaid * 30) / 100;  // 30% to swappers
+        uint256 lpFee = amountPaid - hookFee - swapperFee;  // 60% to pool LPs
+
+        // Transfer 10% to hook deployer
+        require(paymentToken.transfer(hookDeployer, hookFee), "Hook deployer fee transfer failed");
+
+        // Distribute 30% to the swappers
+        uint256 perSwapperFee = swapperFee / commitCount;
+        for (uint256 i = commitBacklog.length - commitCount; i < commitBacklog.length; i++) {
+            require(paymentToken.transfer(commitBacklog[i].swapper, perSwapperFee), "Swapper fee transfer failed");
+            emit FeeDistributed(commitBacklog[i].pool, commitBacklog[i].swapper, perSwapperFee);
+        }
+
+        // Distribute 60% to the LPs
+        uint256 perPoolFee = lpFee / commitCount;
+        for (uint256 i = commitBacklog.length - commitCount; i < commitBacklog.length; i++) {
+            require(paymentToken.transfer(commitBacklog[i].pool, perPoolFee), "LP fee transfer failed");  // Simplified; could route to LPs via pool logic
```

Comment:
These tokens would be locked/lost. You can't just transfer tokens to the pool address (since v1, anyway). You'd have to use the DONATION add liquidity method - and check on registration that the pool supports it. same comment about aggregation transfers.